iceoryx_utils/include/iceoryx_utils/internal/lockfree_queue/buffer.hpp

```diff
@@ -0,0 +1,53 @@
+#pragma once
```

Comment:
Yes, can be done. It is only a little helper class as of now, but might get extended with cacheline optimzations later. The question is a more general one: do we really want to have inl files for every little helper template? We are not entirely consistent across the codebase in this regard, but we should strive for an agreement in the long run (other coding rules may also need some team review).

Reply:
Yeah, separating the declaration from the definition is a must for templates as well IMHO.

Reply:
Will do, but I am not convinced it is good practice for helper classes which are not part of the public interface (i.e. internal hpp files). This will clutter up the filestructure (ok, this can be mitigated with a good file browser) but more importantly lead to more/longer lines of code which carry not much meaning due to the necessary template and namespace qualifiers in the inl  files.